src/Features/CSharp/Portable/ConvertAnonymousTypeToTuple/CSharpConvertAnonymousTypeToTupleCodeFixProvider.cs

```diff
@@ -0,0 +1,156 @@
+                    continue;
+                }
+
+                ReplaceWithTuple(editor, node);
+            }
+
+            return SpecializedTasks.EmptyTask;
+        }
+
+        private static void ReplaceWithTuple(SyntaxEditor editor, AnonymousObjectCreationExpressionSyntax node)
+            => editor.ReplaceNode(
+                node, (current, _) =>
+                {
+                    // Use the callback form as anonymous types may be nested, and we want to
+                    // properly replace them even in that case.
+                    var anonCreation = current as AnonymousObjectCreationExpressionSyntax;
+                    if (anonCreation == null)
+                    {
+                        return current;
+                    }
+
+                    return ConvertToTuple(anonCreation).WithAdditionalAnnotations(Formatter.Annotation);
+                });
+
+        private static AnonymousObjectCreationExpressionSyntax TryGetCreationNode(Diagnostic diagnostic, CancellationToken cancellationToken)
+            => diagnostic.Location.FindToken(cancellationToken).Parent as AnonymousObjectCreationExpressionSyntax;
+
+        private static TupleExpressionSyntax ConvertToTuple(AnonymousObjectCreationExpressionSyntax anonCreation)
+            => SyntaxFactory.TupleExpression(
+                    SyntaxFactory.Token(SyntaxKind.OpenParenToken).WithTriviaFrom(anonCreation.OpenBraceToken),
+                    ConvertInitializers(anonCreation.Initializers),
+                    SyntaxFactory.Token(SyntaxKind.CloseParenToken).WithTriviaFrom(anonCreation.CloseBraceToken))
+                            .WithPrependedLeadingTrivia(anonCreation.GetLeadingTrivia());
+
+        private static SeparatedSyntaxList<ArgumentSyntax> ConvertInitializers(SeparatedSyntaxList<AnonymousObjectMemberDeclaratorSyntax> initializers)
+            => SyntaxFactory.SeparatedList(initializers.Select(ConvertInitializer), initializers.GetSeparators());
+
+        private static ArgumentSyntax ConvertInitializer(AnonymousObjectMemberDeclaratorSyntax declarator)
+            => SyntaxFactory.Argument(ConvertName(declarator.NameEquals), default, declarator.Expression)
+                            .WithTriviaFrom(declarator);
+
+        private static NameColonSyntax ConvertName(NameEqualsSyntax nameEquals)
+            => nameEquals == null
+                ? null
+                : SyntaxFactory.NameColon(
+                    nameEquals.Name,
+                    SyntaxFactory.Token(SyntaxKind.ColonToken).WithTriviaFrom(nameEquals.EqualsToken));
+
+        private class MyCodeAction : CodeAction.DocumentChangeAction
+        {
+            public MyCodeAction(Func<CancellationToken, Task<Document>> createChangedDocument)
+                : base(FeaturesResources.Convert_anonymous_type_to_tuple, createChangedDocument, FeaturesResources.Convert_anonymous_type_to_tuple)
+            {
+            }
+        }
+    }
+}
```

Comment:
I took a quick look (looks good). I'll take a deeper look once the tests are added. VB seems fine to have its own PR.